java/ql/test/library-tests/dataflow/collections/Test.java

```diff
@@ -25,5 +25,21 @@ public void run() {
     Iterator<String> it = m.values().iterator();
     String x5 = it.next();
     sink(x5); // Flow
+
+    it.forEachRemaining(x6 -> {
+      sink(x6); // Flow
+    });
+
+    m.forEach((key, value) -> {
+      sink(key); // Flow
```

Comment:
I notice that `entry.getKey()` isn't considered tainted. Should the same be true here?
If so, is there a good way to know if the key is tainted by user supplied data or not to decide whether or not to filter it out?